source/system/native/MutexUnix.ooc

```diff
@@ -19,7 +19,17 @@ version(unix || apple) {
 			super()
 		}
 		lock: override func {
-			pthread_mutex_lock(this _backend&)
+			version(debugDeadlock) {
+				timeLimitMilliseconds := 1000 //Only whole seconds
```

Comment:
I see. But I disagree, that's a simple if-statement, and if we're at the level of precision where the variable specifies milliseconds, not allowing for fractions of seconds is an unexpected limitation.

```
(deadline tv_sec, deadline tv_nsec) = (currentTime tv_sec + timeLimitMilliseconds / 1000, currentTime tv_usec * 1000 + (timeLimitMilliseconds % 1000) * 1_000_000)
if (deadline tv_nsec >= 1_000_000_000)
	(deadline tv_sec, deadline tv_nsec) = (deadline tv_sec + 1, deadline tv_nsec - 1_000_000_000)
```